src/lxc/tools/lxc_unshare.c

```diff
@@ -241,12 +254,43 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 
+	if (start_arg.setuid) {
+		wait_fd = eventfd(0, EFD_CLOEXEC);
+		if (wait_fd == -1) {
```

Comment:
As I said, above, declare `wait_fd` at the top of main and pass `wait_fd` as arg in start_arg.